app/src/main/java/com/infomaniak/drive/utils/PreviewUtils.kt

```diff
@@ -161,3 +168,39 @@ private fun Context.printPdf(
         file != null -> printFile()
     }
 }
+
+fun downloadFile(
+    externalOutputFile: IOFile,
+    fileModel: File,
+    shouldBePdf: Boolean,
+    onProgress: (progress: Int) -> Unit,
+) {
+    if (externalOutputFile.exists()) externalOutputFile.delete()
+
+    val downloadUrl = fileModel.downloadUrl() + if (fileModel.isOnlyOfficePreview()) "?as=pdf" else ""
```

Comment:
Maybe we can construct the `downloadUrl` in the `File` class itself ?

```kotlin
fun downloadUrl(): String {
   val url = if (isPublicShared()) {
       ApiRoutes.downloadShareLinkFile(driveId, externalShareLinkUuid, file = this)
   } else {
       ApiRoutes.downloadFile(file = this)
   }
   return if (isOnlyOfficePreview()) "$url?as=pdf" else url
}
```

Because it's a bit weird to have this url built in a `PreviewUtils` file.

Reply:
Move to `ApiRoutes`

Reply:
I'll move all the url computing functions of File.kt in another PR dedicated to it